src/solvers/CMakeLists.txt

```diff
@@ -88,10 +88,10 @@ elseif("${sat_impl}" STREQUAL "glucose")
     message(STATUS "Building solvers with glucose")
 
     download_project(PROJ glucose
-        URL http://www.labri.fr/perso/lsimon/downloads/softwares/glucose-syrup.tgz
+        URL https://github.com/BrunoDutertre/glucose-syrup/archive/refs/heads/main.tar.gz
         PATCH_COMMAND patch -p1 -i ${CBMC_SOURCE_DIR}/../scripts/glucose-syrup-patch
         COMMAND cmake -E copy ${CBMC_SOURCE_DIR}/../scripts/glucose_CMakeLists.txt CMakeLists.txt
-        URL_MD5 b6f040a6c28f011f3be994663338f548
+        URL_MD5 89a5183dd9b9eb5296fd9fab2c968a50
```

Comment:
As far as I understand we are downloading the `main` branch, which seems to be something like a `master` branch of sorts.

How often is this being updated, and could random updates cause CI to fail by essence of this hash check failing?

Reply:
The repo only has total of 12 commits, and the most recent commit was July 2021. I would expect the rate of updates to be very low, if any at all. So, yes, CI could fail, but when it does, it is right to do so: it would be a change that we should manually review to make sure it doesn't conflict with any of our expectations on this third-party component.

Reply:
I am not entirely convinced that we should be following the main branch on someone else's fork. Depending on the development workflow they are using, we could be fetching untested or even non-compiling commits. Is there a way we can pin this to a release version?

Reply:
An option would be to get
https://github.com/BrunoDutertre/glucose-syrup/archive/0bb2afd3b9baace6981cbb8b4a1c7683c44968b7.zip

Reply:
Is it worth stepping back a pace or two and consider whether using a submodule or subtree is appropriate? Also other fleeting thoughts... should we contribute our local patches back to Bruno to avoid maintaining them ourselves (and maintaining our local patch management)? Should we fork Bruno's repo, apply our patches there, and download from our fork? No major preferences from my side, but just feels worth a thought and explicit decisions.

Reply:
submodules are tedious to deal with when your build environment doesn't have network access.

Reply:
Given that none of the glucose-syrup repositories have done a release, the only option seems to be what @kroening suggested. That would ensure stability, at the expense of not picking up future changes should they ever occur. Given that we are talking about a piece of software that is unlikely to see further development, maybe that's an ok price to pay?

Reply:
- Fetching the archive of a specific commit seems like a better option than fetching the main branch to me.
- Downloading dependencies as part of the run of `make` / `cmake` vs checking out a submodule seems like moving the problem of needing network access around, rather than solving it to me. Therefore I am not going to argue in favour of one or the other.
- My understanding is that setting up a _subtree_ would mean that the dependency would be included in downloaded archives and initial checkouts of the repository. I can look into this possibility as an alternative to confirm if this is the case.

Reply:
I can confirm that if we were to setup glucose as a subtree, then it would be checked out by default and it would be included in archive downloads from github - https://github.com/thomasspriggs/cbmc/tree/tas/glucose-subtree

Reply:
My preference would be to fork the repo and manage our patches there. I don't think there's anything to upstream to the original repo in our patches, though.
I think the discussion around not having network access in the build environment is not relevant - you already need network access to download minisat in the current setup. Unless this is a new requirement that we must satisfy I don't think subtree is a suitable option for that reason. 
I'm fine with submodules, but I think they are an overkill for our use case as glucose is optional and the interface of glucose is very stable - i.e. no need to tightly track upgrades.
So, my feeling is that fetching the source zip from a fork is probably the must suitable and easiest solution.

Reply:
Reviving this PR/discussion as we're once again blocked by network access issues to labri.fr. If anyone fancies doing (and maintaining) a fork then that's ok with me. I don't have the resources to do so, and will hence have to stick with what I've proposed in this PR at this point.

Reply:
I have now added a comment that indicates the exact revision and  also states that our CI jobs will fail if "main" ever moves.

Reply:
As we're in this situation once again (see https://github.com/diffblue/cbmc/pull/6986): pinging @NlightNFotis @thomasspriggs @peterschrammel for an update.

Reply:
I've now changed download URL to use the specific commit hash rather than a branch.

Reply:
Looks like the right solution for now. Thanks.